ros/lib/utils.py

```diff
@@ -268,7 +268,12 @@ def system_allowed_in_ros(msg, reporter):
         cloud_provider = msg["results"]["system"]["metadata"].get('cloud_provider')
     elif reporter == 'INVENTORY EVENTS':
         cloud_provider = msg['host']['system_profile'].get('cloud_provider')
-        if msg.get('type') == 'updated':
+        # Covers a scenario when ROS is enabled for an existing system on Inventory
+        # i.e. msg type is 'updated' and platform_metadata is available
+        if (
+                msg.get('type') == 'updated'
+                and msg.get('platform_metadata') is None
+        ):
             updated_event = True
```

Comment:
For better understanding, could you rename variable `updated_event' to `updated_event_via_api`?

Reply:
Btw, I didn't understand why cloud_provider check is ONLY applicable for updated event?

Reply:
Ah dang, the cloud_provider check should not be skipped, changes pushed